components/search.json

```diff
@@ -0,0 +1,28 @@
+{
+  "search": {
+    "#rest": {
```

Comment:
It looks like what you've done here is copy the Figma exactly. The #rest, #typing and #typed states aren't real UI states mapping to real UI attributes, I think we need to look more carefully at how to build this component.

I believe it's something more like
`
{
  "search": {
    "placeholder-text": "@theme-text-secondary-normal",
    "#normal": {
      "text:" "@theme-text-primary-normal",
      "background": "@theme-background-solid-primary-normal",
      "border": "@theme-outline-input-normal"
    },
    "#active": {
      "text:" "@theme-text-primary-normal",
      "background": "@theme-background-solid-primary-normal",
      "border": "@theme-outline-input-active"
    },
    "#disabled": {
      "text:" "@theme-text-primary-normal",
      "background": "@theme-background-primary-disabled",
      "text": "@theme-text-primary-disabled"
    }
    ...
`
i.e. the typing and typed states are actually just distinguishing between the placeholder text and value text colours.

Reply:
that makes sense to me as well, I had discussed this with Rekha and she noted that those were not true UI states